coalib/processes/Processing.py

```diff
@@ -685,6 +690,8 @@ def execute_section(section,
                              a file cache buffer.
     :param log_printer:      The log_printer to warn to.
     :param console_printer:  Object to print messages on the console.
+    :param match_cache:      The cached collected file list of previous
```

Comment:
I think that it is related to section based caching which was a part of the Optimize Caching project a year ago but we are not doing that anymore since it doesn't fit well with the NextGen core. I found the issue related to this https://github.com/coala/coala/issues/2699

I think this PR and the issue can be closed since we are not doing section based caching anymore and Siddharta hasn't been an active contributor to coala for a while so I don't think we can find anyone willing to continue this work.

Also introduction of this new arg might have conflicts with my changes to Processing.py so I think it should be closed.